File: implementations/python-fastapi-pragmatic/src/presentation/api/v1/external.py
import asyncio
import time

from fastapi import APIRouter

from src.presentation.schemas.common import ExternalResponse

router = APIRouter(tags=["external"])


@router.get("/external", response_model=ExternalResponse)
async def call_external_api():
    """External API call simulation - async I/O performance"""
    start = time.perf_counter()

    # Simulate external API latency (100ms)
    await asyncio.sleep(0.1)

    latency = (time.perf_counter() - start) * 1000

    return ExternalResponse(
        source="simulated_external_api",
        latency_ms=round(latency, 2),
        data={"message": "External API response"},
    )

File: implementations/typescript-express/src/routes/external.ts
import { Router } from "express";

const router = Router();

// 외부 API 호출 시뮬레이션 (100ms 지연)
router.get("/", async (req, res) => {
  const start = performance.now();

  await new Promise((resolve) => setTimeout(resolve, 100));

  const latency = performance.now() - start;

  res.json({
    source: "simulated_external_api",
    latency_ms: Math.round(latency * 100) / 100, // 소수점 2자리
    data: { message: "External API response" },
  });
});

export default router;
